src/stores/spaces/SpaceStore.ts

```diff
@@ -605,23 +605,22 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
     private switchToRelatedSpace = (roomId: string) => {
         if (this.suggestedRooms.find(r => r.room_id === roomId)) return;
 
-        let parent = this.getCanonicalParent(roomId);
+        // try to find the canonical parent first
+        let parent: SpaceKey = this.getCanonicalParent(roomId)?.roomId;
+
+        // otherwise, try to find a root space which contains this room
         if (!parent) {
-            parent = this.rootSpaces.find(s => this.spaceFilteredRooms.get(s.roomId)?.has(roomId));
+            parent = this.rootSpaces.find(s => this.spaceFilteredRooms.get(s.roomId)?.has(roomId))?.roomId;
         }
+
+        // otherwise, try to find a metaspace which contains this room
         if (!parent) {
-            const parentIds = Array.from(this.parentMap.get(roomId) || []);
-            for (const parentId of parentIds) {
-                const room = this.matrixClient.getRoom(parentId);
-                if (room) {
-                    parent = room;
-                    break;
-                }
-            }
+            // search meta spaces in reverse as Home is the first and least specific one
+            parent = [...this.enabledMetaSpaces].reverse().find(s => this.getSpaceFilteredRoomIds(s).has(roomId));
         }
 
         // don't trigger a context switch when we are switching a space to match the chosen room
-        this.setActiveSpace(parent?.roomId ?? MetaSpace.Home, false); // TODO
+        this.setActiveSpace(parent ?? MetaSpace.Home, false); // TODO
```

Comment:
Is "TODO" still relevant here?

Reply:
Yeah because MetaSpace.Home may be disabled so its an unhandled (Labs) edge-case